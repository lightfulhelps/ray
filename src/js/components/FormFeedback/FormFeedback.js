// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.Node,
  className?: string,
  isValid?: boolean,
  tag?: string,
};

const FormFeedback = ({ children, className, isValid, tag: Tag = 'div', ...other }: Props) => {
  const classes = classNames(className, isValid ? 'valid-feedback' : 'invalid-feedback');

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default FormFeedback;
