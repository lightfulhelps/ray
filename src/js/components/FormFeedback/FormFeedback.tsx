import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ReactNode;
  className?: string;
  isValid?: boolean;
  tag?: keyof JSX.IntrinsicElements;
  [key: string]: any;
};

const FormFeedback: React.FC<Props> = ({
  children,
  className,
  isValid,
  tag: Tag = 'div',
  ...other
}: Props) => {
  const classes = classNames(className, isValid ? 'valid-feedback' : 'invalid-feedback');

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default FormFeedback;
