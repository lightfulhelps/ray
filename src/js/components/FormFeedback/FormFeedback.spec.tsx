import React from "react";
import { shallow } from "enzyme";
import merge from "lodash/merge";
import FormFeedback from "./FormFeedback";

const setup = (overrides = {}) => {
  const props = merge({
    children: 'Invalid value for...'
  }, overrides);
  const wrapper = shallow(<FormFeedback {...props} />);

  return { wrapper, props };
};

describe('<FormFeedback />', () => {
  it('should render', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle children', () => {
    const {
      wrapper
    } = setup({ children: 'Children' });

    expect(wrapper.text()).toBe('Children');
  });

  it('should handle className', () => {
    const {
      wrapper
    } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('invalid-feedback')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle isValid', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.hasClass('invalid-feedback')).toBe(true);
    expect(wrapper.hasClass('valid-feedback')).toBe(false);

    wrapper.setProps({ isValid: true });

    expect(wrapper.hasClass('invalid-feedback')).toBe(false);
    expect(wrapper.hasClass('valid-feedback')).toBe(true);
  });

  it('should handle tag', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.type()).toBe('div');

    wrapper.setProps({ tag: 'span' });

    expect(wrapper.type()).toBe('span');
  });

  it('should pass through other props', () => {
    const {
      wrapper
    } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});