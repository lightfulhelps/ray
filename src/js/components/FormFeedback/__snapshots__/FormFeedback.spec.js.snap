// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFeedback /> should render 1`] = `
<div
  className="invalid-feedback"
>
  Invalid value for...
</div>
`;
